File: src/main/java/liqp/filters/Split.java
package liqp.filters;

import java.util.regex.Pattern;
import liqp.nodes.RenderContext;

class Split extends Filter {

    /*
     * split(input, delimiter = ' ')
     *
     * Split a string on a matching pattern
     *
     * E.g. {{ "a~b" | split:'~' | first }} #=> 'a'
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        String original = super.asString(value);

        String delimiter = super.asString(super.get(0, params));

        return original.split("(?<!^)" + Pattern.quote(delimiter));
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Split.kt
package liqp.filter

import liqp.context.LContext
import java.util.regex.Pattern

class Split : LFilter() {

  /**
   * split(input, delimiter = ' ')
   *
   * Split a string on a matching pattern
   *
   * E.g. {{ "a~b" | split:'~' | first }} #=> 'a'
   */
  override fun onFilterAction(params: FilterParams, value: Any?,
                              context: LContext): Any? {

    val original = context.asString(value)
    val delimiter = context.asString(params[0]) ?: ""

    return original
        ?.split(("(?<!^)" + Pattern.quote(delimiter)).toRegex())
        ?.dropLastWhile { it.isEmpty() }
  }
}
